cluster/local/provider.Tiltfile

```diff
@@ -0,0 +1,196 @@
+####################################################################################
+# Crossplane Providers helper functions
+#
+# These are helper functions to build, deploy, and live-reload providers.
+# Note that these functions are shared with both:
+#
+# 1) core crossplane repo
+# 2) any standalone provider
+####################################################################################
+
+# Register CRDs which expose `image` for Tilt to be able to find them and
+# build the corresponding docker image for it.
+# https://docs.tilt.dev/custom_resource.html#does-it-contain-images
+k8s_kind('ControllerConfig', image_json_path = '{.spec.image}')
+
+# Build a crossplane provider
+#
+# 1. build and apply CRDs
+# 2. build and deploy controller manager
+def build_provider(name, provider, settings):
+    _generate_crds(provider, settings)
+    _build_controller(name, provider, settings)
+
+# Generate and apply crossplane provider's CRDs
```

Comment:
This may no longer be needed as Crossplane has an `initContainer` that manages CRDs.

Reply:
Good to know, I haven't tested this recently but one thing gives me pause (just by looking at the code) is that you say this runs in `initContainer` but the `provider.Tiltfile` code (as it stands) doesn't care about it and even when the Tilt is running and if it detects a change in `api/` folder it will regenerate and reapply CRDs anyways (as opposed to one-off operations of `initContainer`).

But for sure this can be revisited and if we can remove it, all the better!